### PR TITLE
Forecast hotfix for accessing BigQuery data

### PR DESCRIPTION
**Pull request by Mnoble-19**
**_WHAT DOES THIS PR DO?_**
* This PR adds the gsc credentials parameter when accessing BigQuery data
**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - []()
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**
Screenshot 2023-04-13 at 13 51 35

### CODE REVIEW

**Comment by Baalmart** [comment]
Hi @Mnoble-19 , thanks for the amazing work!

1. It shows that the **end point does not exist on my end**. Care to test out too?
2. Also, I hope we are also able to do some input validations for bad requests, For my case, I pass a wrong site_id
Screenshot 2023-04-15 at 15 02 00

**Comment by Baalmart** [comment]
Additionally @Mnoble-19 , what if we just organising an online working meeting....this can be for an entire week day so that we can get this forecast thing completely out of the way?  cc: @sserurich 

Thanks!

**Comment by Mnoble-19** [comment]
> Hi @Mnoble-19 , thanks for the amazing work!
> 
> 1. It shows that the **end point does not exist on my end**. Care to test out too?
> 2. Also, I hope we are also able to do some input validations for bad requests, For my case, I pass a wrong site_id
> 
> Screenshot 2023-04-15 at 15 02 00

Hello @Baalmart this failed because you're using a wrong endpoint. Paste this [link](https://forecasts-hotfixes-predict-service-preview-w7kzhvlewq-ew.a.run.app/api/v2/daily-forecast?site_id=60d058c8048305120d2d6140) in your personal workspace. I haven't added the endpoint to postman. 

Regarding the bad requests, there was some validation incase the parameter used in not a string, but for this case where a wrong site is put, just an  empty result is returned. Let me add a more appropriate response
Screenshot 2023-04-15 at 15 28 23

**Comment by Mnoble-19** [comment]
> Additionally @Mnoble-19 , what if we just organising an online working meeting....this can be for an entire week day so that we can get this forecast thing completely out of the way? cc: @sserurich
> 
> Thanks!

This PR is the last step actually, i.e upon merge there will be  an endpoint for the mobile app. 😅
Just waiting for a green flag from @sserurich , and we're good to go.

File: src/predict/api/helpers/utils.py [hunk 1]
```diff
@@ -71,6 +71,10 @@ def get_forecasts_helper(db_name):
     """
     if request.method == 'GET':
         site_id = request.args.get('site_id')
+        if site_id is None or not isinstance(site_id, str):
+            return jsonify({"message": "Please specify a site_id", "success": False}), 400
+        if len(site_id) != 24:
+            return jsonify({"message": "Please enter a valid site_id", "success": False}), 400
```
Comment (Mnoble-19) [inline src/predict/api/helpers/utils.py:77]: @Baalmart this adds checks for invalid inputs

File: src/predict/jobs/forecast_training/transform.py [hunk 2]
```diff
@@ -106,3 +108,14 @@ def get_boundary_layer_data():
 
 def get_forecast_data():
     return _transform_events(Events().get_events_db())
+
+
+def fetch_bigquery_data():
+    """gets data from the bigquery table"""
+    credentials = service_account.Credentials.from_service_account_file(configuration.CREDENTIALS)
+    query = f"""
+    SELECT DISTINCT timestamp, site_id, device_number,pm2_5_calibrated_value FROM `{configuration.GOOGLE_CLOUD_PROJECT_ID}.averaged_data.hourly_device_measurements` where DATE(timestamp) >= DATE_SUB(CURRENT_DATE(), INTERVAL 12 MONTH) and tenant = 'airqo' ORDER BY timestamp
```
Comment (sserurich) [inline src/predict/jobs/forecast_training/transform.py:117]: @Mnoble-19 , we can make this configurable, but this doesn't prevent merge.

**Review by sserurich — APPROVED** [review]
LGTM

**Review by NoahNsimbe — APPROVED** [review]

**Pull request merged**